calc/active_rules_calculator.go

```diff
@@ -190,12 +195,22 @@ func (arc *ActiveRulesCalculator) OnUpdate(update api.Update) (_ bool) {
 				log.Debug("Policy updated while active, telling listener")
 				arc.sendPolicyUpdate(key)
 			}
+
+			// update ALP policies set.
+			if arc.isALPPolicy(policy) {
+				arc.allALPPolicies.Add(key)
+			}
```

Comment:
Need an else here, in case the policy was updated from AL to non-AL

Reply:
Yes. Done.